src/app/clusters/service-area-server/service-area-server.cpp

```diff
@@ -0,0 +1,1101 @@
+#include <app-common/zap-generated/ids/Commands.h>
+#include <app/AttributeAccessInterfaceRegistry.h>
+#include <app/CommandHandler.h>
+#include <app/ConcreteCommandPath.h>
+#include <app/EventLogging.h>
+#include <app/InteractionModelEngine.h>
+#include <app/reporting/reporting.h>
+#include <app/util/attribute-storage.h>
+#include <app/util/util.h>
+#include <lib/support/CodeUtils.h>
+#include <lib/support/logging/CHIPLogging.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+void MatterServiceAreaPluginServerInitCallback(){};
+
+using namespace chip;
+using namespace chip::app;
+using namespace chip::app::Clusters;
+using namespace chip::app::Clusters::ServiceArea;
+using namespace chip::app::Clusters::ServiceArea::Commands;
+using namespace chip::app::Clusters::ServiceArea::Attributes;
+using Status = Protocols::InteractionModel::Status;
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace ServiceArea {
+
+// ****************************************************************************
+// Service Area Server Instance
+
+Instance::Instance(Delegate * aDelegate, EndpointId aEndpointId, BitMask<ServiceArea::Feature> aFeature) :
+    AttributeAccessInterface(MakeOptional(aEndpointId), Clusters::ServiceArea::Id),
+    CommandHandlerInterface(MakeOptional(aEndpointId), Clusters::ServiceArea::Id), mDelegate(aDelegate), mEndpointId(aEndpointId),
+    mClusterId(Clusters::ServiceArea::Id), mFeature(aFeature)
+{
+    ChipLogProgress(Zcl, "Service Area: Instance constructor");
+    mDelegate->SetInstance(this);
+}
+
+Instance::~Instance()
+{
+    InteractionModelEngine::GetInstance()->UnregisterCommandHandler(this);
+    unregisterAttributeAccessOverride(this);
+}
+
+CHIP_ERROR Instance::Init()
+{
+    ChipLogProgress(Zcl, "Service Area: INIT");
+
+    // Check if the cluster has been selected in zap
+    VerifyOrReturnError(emberAfContainsServer(mEndpointId, chip::app::Clusters::ServiceArea::Id), CHIP_ERROR_INVALID_ARGUMENT,
+                        ChipLogError(Zcl, "Service Area: The cluster with Id %lu was not enabled in zap.",
+                                     long(chip::app::Clusters::ServiceArea::Id)));
+
+    ReturnErrorOnFailure(InteractionModelEngine::GetInstance()->RegisterCommandHandler(this));
+
+    VerifyOrReturnError(registerAttributeAccessOverride(this), CHIP_ERROR_INCORRECT_STATE);
+
+    ReturnErrorOnFailure(mDelegate->Init());
+
+    return CHIP_NO_ERROR;
+}
+
+//*************************************************************************
+// core functions
+
+CHIP_ERROR Instance::Read(const ConcreteReadAttributePath & aPath, AttributeValueEncoder & aEncoder)
+{
+    ChipLogError(Zcl, "Service Area: Reading");
+
+    switch (aPath.mAttributeId)
+    {
+
+    case ServiceArea::Attributes::SupportedLocations::Id: {
+        ChipLogProgress(Zcl, "Service Area: Read SupportedLocations");
+        ReturnErrorOnFailure(ReadSupportedLocations(aEncoder));
+        break;
+    }
```

Comment:
```suggestion
    case ServiceArea::Attributes::SupportedLocations::Id:
        ChipLogProgress(Zcl, "Service Area: Read SupportedLocations");
        return ReadSupportedLocations(aEncoder);
```

Applies throughout: return directly (reduce scope of things), remove extra `{}`.
Also wondering if the logs are really that desirable.

Reply:
Addressed [here](https://github.com/project-chip/connectedhomeip/pull/33991/commits/2ab2f933c75b6a385a146a18b77d6da5ba16e0bf).